keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,903 @@
+    // by workloads
+    BucketReady bool
+
+    // BucketName is the name of the provisioned Bucket in response
+    // to this BucketClaim. It is generated and set by the COSI controller 
+    // before making the creation request to the OSP backend.
+    // +optional
+    BucketName string
+  }
+```
+## BucketClass
+
+Resouce for configuring common properties for multiple Buckets. BucketClass is cluster-scoped.
 
-message ProvisionerDeleteBucketResponse {
-     // Intentionally left blank
+```yaml
+BucketClass {
+  TypeMeta
+  ObjectMeta
+
+  // DriverName is the name of driver associated with this bucket
+  DriverName string
+
+  // DeletionPolicy is used to specify how COSI should handle deletion of this
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  DeletionPolicy DeletionPolicy
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisionerGrantBucketAccess
+## BucketAccess
 
-This call grants access to a particular principal. The principal is the account for which this access should be granted. 
+A resource to access a Bucket. BucketAccess is namespace-scoped
 
-If the principal is set, then it should be used as the username of the created credentials or in some way should be deterministically used to generate a new credential for this request. This principal will be used as the unique identifier for deleting this access by calling ProvisionerRevokeBucketAccess
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketAccessSpec {
+    // BucketClaimName is the name of the BucketClaim.
+    // Exactly one of BucketClaimName or BucketName must be set.
+    BucketClaimName string
+
+    // Protocol is the name of the Protocol 
+    // that this access credential is supposed to support
+    // If left empty, it will choose the protocol supported
+    // by the bucket. If the bucket supports multiple protocols,
+    // the end protocol is determined by the driver. 
+    // +optional
+    Protocol Protocol
+
+    // BucketAccessClassName is the name of the BucketAccessClass
+    BucketAccessClassName string
+
+    // CredentialsSecretName is the name of the secret that COSI should populate
+    // with the credentials. If a secret by this name already exists, then it is
+    // assumed that credentials have already been generated. It is not overridden.
+    // This secret is deleted when the BucketAccess is delted.
+    CredentialsSecretName string
+    
+    // ServiceAccountName is the name of the serviceAccount that COSI will map
+    // to the OSP service account when IAM styled authentication is specified
+    ServiceAccountName string
```

Comment:
If I understand correctly, `AuthenticationType: KEY` does not use a ServiceAccount. Should this be optional?

Reply:
That is correct. I'll update it